ui/pages/confirm-add-suggested-token/confirm-add-suggested-token.js

```diff
@@ -117,11 +118,18 @@ const ConfirmAddSuggestedToken = (props) => {
     );
   }, [suggestedAssets, tokens, t]);
 
+  useEffect(() => {
+    if (Object.keys(addedToken).length) {
+      trackTokenAddedEvent();
+    }
+  }, [addedToken, trackTokenAddedEvent]);
+
   useEffect(() => {
     if (!suggestedAssets.length) {
       history.push(mostRecentOverviewPage);
     }
-  }, [history, suggestedAssets, mostRecentOverviewPage]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why was `// eslint-disable-next-line react-hooks/exhaustive-deps` needed?

Reply:
Passing an empty array will ensure this hook is only called once. This is the closest I think we could come to replicating the `componentDidMount` which is what was used when this was a class component. 

see "Note:" at the bottom of ref: https://reactjs.org/docs/hooks-effect.html

If we were to pass `suggestedAssets` as a dependent property here, `history.push(mostRecentOverviewPage);` could override other redirects that happen after altering the `suggestedAssets`. For instance, in the `ConfirmImportToken` page, we redirect the user to the token page after the token has been added. If we were to replicate that behavior here, using `useEffect` with `[suggestedAssets]` dependency list would cause a race condition.

It is recommended that we keep the ESLint warning for `react-hooks/exhaustive-deps` and not use this rule as a last resort.

ref: https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often
> Only do this if you couldn’t find a better alternative, as relying on mutation makes components less predictable.

Reply:
Okay, I think you can do something like this:

```
const goBackIfNoSuggestedAssetsOnFirstRender = () => {
    if (!suggestedAssets.length) {
      history.push(mostRecentOverviewPage);
    }
}

useEffect(() => {
    goBackIfNoSuggestedAssetsOnFirstRender();
}, []);
```

Reply:
lgtm! Thanks for the function name suggestion. Updated the repo with this suggestion

Reply:
@danjm 

lgtm! Thanks for the function name suggestion. Updated the repo with this suggestion

Reply:
I think `// eslint-disable-next-line react-hooks/exhaustive-deps` can be removed now?

Reply:
@danjm @adonesky1 

Removing `// eslint-disable-next-line react-hooks/exhaustive-deps` will enable the warning message again on `[]`.  Adding `goBackIfNoSuggestedAssetsOnFirstRender` as a dependency would cause useEffect to rerun again on each render unless we wrapped it in useCallback, or declare the function inside the useEffect hook. With either of these options, we would run into the need for the empty dependency list, `[]`, either on the useCallback hook or the useEffect hook. 

Is there another way we could go about removing `// eslint-disable-next-line react-hooks/exhaustive-deps` and only rendering this check once on initial load? 

ESLint warning after replacing `[]` with `[goBackIfNoSuggestedAssetsOnFirstRender]` in useEffect:
<img width="1065" alt="Screenshot 2022-03-03 at 1 20 56 PM" src="https://user-images.githubusercontent.com/20778143/156606417-f7896187-aee5-466c-9a59-2dd9bc63e267.png">

Reply:
Does it not work to wrap `goBackIfNoSuggestedAssetsOnFirstRender` in `useCallback` add all dependencies to the dependency array and then add `goBackIfNoSuggestedAssetsOnFirstRender` to the dependency array of the `useEffect`?

Reply:
I don't think so. The initial logic lives in `componentDidMount`, so the thing is, we only want to run this logic on initial load or initial render in this case. I think with this suggestion, if the list of suggestions changes, for instance on `acceptWatchAsset`, then this `useEffect` will run because one of the dependencies of `useCallback` has changed. This causes a race condition between `history.push` here, and the one in the `Add Tokens` click handler.

Reply:
@adonesky1 

I don't think so. The initial logic lives in componentDidMount, so the thing is, we only want to run this logic on initial load or initial render in this case. I think with this suggestion, if the list of suggestions changes, for instance on acceptWatchAsset, then this useEffect will run because one of the dependencies of useCallback has changed. This causes a race condition between history.push here, and the one in the Add Tokens click handler.

Reply:
I can think of an edge case here, what is `suggestedAssets` are not loaded on first render ?

Reply:
@jpuri 

This is a good question. 

What I see happening is when the `wallet_watchAsset` [API request](https://docs.metamask.io/guide/registering-your-token.html) is requested, the `watchAssetHandler` RPC middleware calls [TokensController#watchAsset](https://github.com/MetaMask/controllers/blob/main/src/assets/TokensController.ts#L393). This method validates the asset, updates the redux `suggestedAssets` state with the new asset, and then emits `pendingSuggestedAsset`. `pendingSuggestedAsset` will then call [background.js#openPopup](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/background.js#L605). Unless I am missing something, I think we can safely assume that it will be there before the popup is opened. 

I also think it is safe to make this change due to the order that the hooks run.  `componentDidMount` runs after mounting and before painting. `useEffect` runs after the first paint. We had replaced `componentDidMount` in https://github.com/MetaMask/metamask-extension/pull/13377/files, so the logic is near the same here.